common-cdi/src/main/java/fi/metatavu/edelphi/panels/PanelController.java

```diff
@@ -87,15 +104,20 @@ public class PanelController {
    * @return panel or null if not found
    */
   public Panel findPanelById(Long id) {
-    return panelDAO.findById(id);
+    Panel panel = panelDAO.findById(id);
+    if (panel != null && panel.getArchived()) {
```

Comment:
Archived seems to be checked on the calling side, so maybe you should just double-check that every time you call this method it is handled appropriately instead of checking archived here